### DISCUSSION TITLE
Error: cannot find symbol import com.swmansion.rnscreens.RNScreensPackage.

### DISCUSSION THREAD

**Discussion by dianamgalindo** [software-mansion/react-native-screens#2101] [Q&A]
Hello everyone ! 

I am having issues with my app not building when I install react-native-screens. I am currently receiving two errors that relate to not found symbols: 

<img width="850" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/95235340/059034fa-c34e-46ff-99d4-e1e5538b5f1a">

I have followed the[ installation guide ](https://reactnavigation.org/docs/getting-started) and I have edited my `MainActivity.kt` file with the recommended Kotlin code, as well as adding `import android.os.Bundle;` to the top of the file.
```
class MainActivity: ReactActivity() {
  // ...
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(null)
  }
  // ...
}
```

I am not entirely sure how to fix this. I haven't seen many people having similar issues with installing react-native-screens.
**Here are the versions of the dependencies I am currently using:**

```
"dependencies": {
        "@react-navigation/native": "^6.1.17",
        "@react-navigation/native-stack": "^6.9.26",
        "@react-navigation/stack": "^6.3.29",
        "react": "18.2.0",
        "react-native": "0.73.6",
        "react-native-paper": "^5.12.3",
        "react-native-safe-area-context": "^4.9.0",
        "react-native-screens": "^3.30.1",
        "react-native-vector-icons": "^10.0.3",
        "react-native-vision-camera": "^3.9.2",
        "react-router-native": "^6.22.3"
      }
```
**And my build.gradle**
```
buildscript {
    ext {
        buildToolsVersion = "34.0.0"
        minSdkVersion = 21
        compileSdkVersion = 34
        targetSdkVersion = 34
        ndkVersion = "24.0.8215888"
        kotlinVersion = "1.8.0"
    }
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle")
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin")
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
    }
}

apply plugin: "com.facebook.react.rootproject"
```

It would be greatly appreciated to know if you have faced a similar issue and how you came about solving it. 
Thank you for your time! ❤️ 

Diana

**Comment by kkafar** [comment]
Hey, your issue description suggests me that React Native CLI did include  `react-native-screens` into gradle build (didn't add `react-native-screens` as build dependency of your project. 

Would you send me output of `npx react-native config` command run in the root directory of your application? 

Also, do you have anything unusual / custom in your setup, or is this just standard react native application straight from template?

**Comment by malyzkhan** [comment]
I am facing the same issue After i upgraded my RN version: 
my repo: https://github.com/muhammadalizkhan/DariyPureAppHome/

`BUILD FAILED in 2m 28s
error Failed to install the app. Command failed with exit code 1: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081
e: file:///E:/React%20Native%20Production%20Work/DairyPureAppInv/android/app/src/main/java/com/dairypureappinv/MainApplication.kt:13:32 Unresolved reference: RNScreensPackage e: file:///E:/React%20Native%20Production%20Work/DairyPureAppInv/android/app/src/main/java/com/dairypureappinv/MainApplication.kt:22:15 Unresolved reference: RNScreensPackage FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':app:compileDebugKotlin'. > A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction > Compilation error. See log for more details * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. > Get more help at https://help.gradle.org. BUILD FAILED in 2m 28s.
info Run CLI with --verbose flag for more details.`

this is my MainApplication.kt file

```kotlin
package com.dairypureappinv

import android.app.Application
import com.facebook.react.PackageList
import com.facebook.react.ReactApplication
import com.facebook.react.ReactHost
import com.facebook.react.ReactNativeHost
import com.facebook.react.ReactPackage
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.load
import com.facebook.react.defaults.DefaultReactHost.getDefaultReactHost
import com.facebook.react.defaults.DefaultReactNativeHost
import com.facebook.soloader.SoLoader
import com.swmansion.rnscreens.RNScreensPackage

class MainApplication : Application(), ReactApplication {

  override val reactNativeHost: ReactNativeHost =
    object : DefaultReactNativeHost(this) {
      override fun getPackages(): List<ReactPackage> =
        PackageList(this).packages.apply {
          // Add RNScreensPackage here for explicit linking (optional)
          add(RNScreensPackage())
        }

      override fun getJSMainModuleName(): String = "index"

      override fun getUseDeveloperSupport(): Boolean = BuildConfig.DEBUG

      override val isNewArchEnabled: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
      override val isHermesEnabled: Boolean = BuildConfig.IS_HERMES_ENABLED
    }

  override val reactHost: ReactHost
    get() = getDefaultReactHost(applicationContext, reactNativeHost)

  override fun onCreate() {
    super.onCreate()
    SoLoader.init(this, false)
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      // If you opted-in for the New Architecture, we load the native entry point for this app.
      load()
    }
  }
}
```

**Reply by leoderigo** [reply]
Did you found a solution?

**Comment by Omkar-Jawalkar** [comment]
Same problem here!
Any solutions available?

**Comment by arunvc** [comment]
Try this, worked for me after upgrade to targetSdkVersion = 34

delete node_modules
delete package-lock.json

````
npm i 
or
npm i --legacy-peer-deps

````

cd android
./gradlew clean

**Comment by KrisLau** [comment]
Did anyone manage to fix it?

**Comment by ramakrushna-digitivity-developer** [comment]
Same issue

**Comment by kkafar** [comment]
If this is an real issue for any one of you, please open an appropriate ticket with reproduction at https://github.com/software-mansion/react-native-screens/issues.
